src/ResourceManager/IotHub/Commands.IotHub/IotHub/SetAzureRmIotHub.cs

```diff
@@ -122,6 +126,27 @@ public class SetAzureRmIotHub : IotHubBaseCmdlet
         [ValidateNotNullOrEmpty]
         public PSOperationsMonitoringProperties OperationsMonitoringProperties { get; set; }
 
+        [Parameter(
+            ParameterSetName = UpdateRoutingPropertiesParameterSet,
+            Mandatory = false,
+            HelpMessage = "RoutingProperties")]
+        [ValidateNotNullOrEmpty]
+        public PSRoutingProperties RoutingProperties { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateRoutePropertiesParameterSet,
+            Mandatory = false,
+            HelpMessage = "Routes")]
+        [ValidateNotNullOrEmpty]
+        public List<PSRouteProperties> Routes { get; set; }
```

Comment:
What is the difference between RouteProperties and routing properties?  You shoiuld just take an array of the type rather than a list.  Also, parameter name should be singular (Route)